railties/lib/rails/generators/rails/controller/controller_generator.rb

```diff
@@ -20,10 +20,20 @@ def add_routes
         route generate_routing_code
       end
 
-      hook_for :template_engine, :test_framework, :helper, :assets
+      hook_for :template_engine, :test_framework, :helper, :assets do |generator|
+        invoke generator, [ remove_possible_suffix(name), actions ]
+      end
 
       private
 
+        def file_name
+          @_file_name ||= remove_possible_suffix(super)
```

Comment:
It might be better to use `#gsub` inline like the other lines you referenced for consistency sake.